sdks/router-sdk/src/entities/trade.ts

```diff
@@ -155,13 +155,49 @@ export class Trade<TInput extends Currency, TOutput extends Currency, TTradeType
 
     const outputCurrency = this.swaps[0].outputAmount.currency
     const totalOutputFromRoutes = this.swaps
-      .map(({ outputAmount }) => outputAmount)
+      .map(({ outputAmount: routeOutputAmount }) => routeOutputAmount)
       .reduce((total, cur) => total.add(cur), CurrencyAmount.fromRawAmount(outputCurrency, 0))
 
     this._outputAmount = totalOutputFromRoutes
     return this._outputAmount
   }
 
+  /**
+   * Returns the sum of all swaps within the trade where the pathInput/Output is native
```

Comment:
```suggestion
   * Returns the sum of all swaps within the trade
```

Reply:
maybe im wrong, but this function returns the sum of all swaps, even when pathInput/Output _isnt_ native?
Its just the 2 `xAmountNative` variables that might not get returned for this case